service/share/share.go

```diff
@@ -144,17 +144,37 @@ func (s *service) GetSharesByNamespace(ctx context.Context, root *Root, nID name
 		return nil, ipld.ErrNotFoundInRange
 	}
 
-	namespacedShares := make([]Share, 0)
+	type res struct {
+		nodes []format.Node
+		err   error
+	}
+	resultCh := make(chan *res)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	for _, rootCID := range rowRootCIDs {
-		nodes, err := ipld.GetLeavesByNamespace(ctx, s.dag, rootCID, nID)
-		if err != nil {
-			return nil, err
-		}
+		go func(rootCID cid.Cid) {
+			nodes, err := ipld.GetLeavesByNamespace(ctx, s.dag, rootCID, nID)
+			resultCh <- &res{nodes: nodes, err: err}
+		}(rootCID)
+	}
 
-		for _, node := range nodes {
-			namespacedShares = append(namespacedShares, node.RawData()[1:])
+	namespacedShares := make([]Share, 0)
+	for i := 0; i < len(rowRootCIDs); i++ {
+		select {
+		case result := <-resultCh:
```

Comment:
Should we check for `ok` as well / make sure`result` is never nil? It could technically panic otherwise, no?

Reply:
`resultCh` is never closed, so that is not an issue

Reply:
and there's no instance where result could ever be nil?

Reply:
Yes, you can see it from the code.